aiohttp/web_fileresponse.py

```diff
@@ -122,29 +130,31 @@ async def _precondition_failed(
         return await super().prepare(request)
 
     def _get_file_path_stat_and_gzip(
-        self, check_for_gzipped_file: bool
-    ) -> Tuple[pathlib.Path, os.stat_result, bool]:
-        """Return the file path, stat result, and gzip status.
+        self, accept_encoding: str
+    ) -> Tuple[pathlib.Path, os.stat_result, Optional[ContentCoding | Literal["br"]]]:
+        """Return the file path, stat result, and possible compression type.
 
         This method should be called from a thread executor
         since it calls os.stat which may block.
         """
         filepath = self._path
-        if check_for_gzipped_file:
```

Comment:
Let’s keep the guard here if the accept encoding isn’t set. It won’t make much difference but we could also avoid the lower on the header if it’s not set.

Reply:
I'm not following - there is no check for no accept-encoding header in the current code.  The logic here I believe is identical to the current code, just a loop over 2 ifs instead of one.

Regarding no `accept-encoding` header in general though:
1. No browser/client does that, and it's a bad idea as it indicates per spec that any coding is acceptable
2. The current code is (arguably but not really) not spec compliant in that sense - it's always sending uncompressed which is not best for the server if the client says anything goes (and it probably has never come up as an issue because of my previous point).  The same goes for dynamic compression here: https://github.com/steverep/aiohttp/blob/b0d58f7c6164bff5ca59fd3e97cfeb652eac3ef5/aiohttp/web_response.py#L330.

So, I would say a follow up could be to adjust that trivial compliance issue, and then you could save the `lower()` call if desired.  However, I would recommend not doing it as you suggested below and instead keeping it within the async code.

Reply:
>  there is no check for no accept-encoding header in the current code. 

`check_for_gzipped_file` previously meant `"gzip" in request.headers.get(hdrs.ACCEPT_ENCODING, "")` so it would not enter this block if there was no `ACCEPT_ENCODING` header

Reply:
Here is the full suggestion https://github.com/aio-libs/aiohttp/commit/6409638c9e83c8bb7efeb86da97228919df471da

Reply:
> 2\. The current code is (arguably but not really) not spec compliant in that sense

I think you are referring to:
> https://www.rfc-editor.org/rfc/rfc9110#section-12.5.3
...
> If no Accept-Encoding header field is in the request, any content coding is considered acceptable by the user agent.

While I agree the current code is not compliant with the RFC, I think if we changed the implementation to work in this way, we would break client that can't handle `br` encoding that fail to send an `Accept-Encoding` header.  

While its RFC correct to break them, but we may get fallout from it, and I wouldn't want to do that in this PR as it is a practical risk we would need to revert a change like that

Reply:
It looks like the common case we need to `lower()` it anyways so this doesn't end up saving anything anyways.